### PR TITLE
Query Response format to be based on http 'accept' header & Query Payload content type to be based on 'content-type' header

### PR DESCRIPTION
**Pull request by jaihind213**
Currently the Query Response format is based on the 'content-type' header, if a user sends a request, both the query payload and response must be in the same format.

This pull request is about having:

Query Response format based on http 'accept' header & Query Payload content type based on 'content-type' header.

so then

the user can send the http query payload in 'application/json' and get a response in 'x-jackson-smile' format instead of 'application/json'.

This makes it easy for the User to send the query payload in friendly json format and get the response in 'x-jackson-smile' i.e. a compressed format. (best of both worlds)

### CODE REVIEW

File: server/src/main/java/io/druid/server/QueryResource.java [hunk 1]
```diff
@@ -167,19 +167,19 @@ public Response getServer(@PathParam("id") String queryId, @Context final HttpSe
...
+      @Context final HttpServletRequest req // used to get request content-type,Accept header, remote address and AuthorizationInfo
   ) throws IOException
   {
     final long start = System.currentTimeMillis();
     Query query = null;
     QueryToolChest toolChest = null;
     String queryId = null;
 
-    final ResponseContext context = createContext(req.getContentType(), pretty != null);
+    final ResponseContext context = createContext(req.getHeader("Accept"), pretty != null);
```
Comment (drcrallen) [inline server/src/main/java/io/druid/server/QueryResource.java]: To retain prior behavior this should default to the content type if the `Accept` header is missing

**Comment by jaihind213** [comment]
hi @drcrallen 

made the changes requested.
o- if Accept header is absent, it defaults to Content-Type header
o- test cases for the same.

**Comment by niketh** [comment]
@jaihind213 Are you planning to finish this up?

**Comment by churrodog** [comment]
any plan on getting this in?  This would be very useful.

**Comment by jihoonson** [comment]
@jaihind213 sorry for the delayed review. The PR looks good to me. Would you please fix the conflicts?

**Comment by jaihind213** [comment]
thanks for the review. will resolve conflict asap.

**Comment by jaihind213** [comment]
hi @jihoonson  - i have resolved merge conflicts.

**Comment by niketh** [comment]
@jihoonson @drcrallen Can we merge this if it looks good.

File: server/src/main/java/org/apache/druid/server/QueryResource.java [hunk 2]
```diff
@@ -156,13 +156,19 @@ public Response cancelQuery(@PathParam("id") String queryId, @Context final Http
...
-      @Context final HttpServletRequest req // used to get request content-type, remote address and auth-related headers
+      @Context final HttpServletRequest req // used to get request content-type,Accept header, remote address and auth-related headers
   ) throws IOException
   {
     final QueryLifecycle queryLifecycle = queryLifecycleFactory.factorize();
     Query<?> query = null;
 
-    final ResponseContext context = createContext(req.getContentType(), pretty != null);
+    String acceptHeader = req.getHeader("Accept");
+    if (acceptHeader == null || acceptHeader.length() == 0) {
```
Comment (drcrallen) [inline server/src/main/java/org/apache/druid/server/QueryResource.java]: (minor) `Strings.isNullOrEmpty` is usually used in such a scenario

**Comment by drcrallen** [comment]
Thanks for your patience!

File: server/src/main/java/org/apache/druid/server/QueryResource.java (hunk 2 above)
Comment (jaihind213) [reply server/src/main/java/org/apache/druid/server/QueryResource.java]: hi @drcrallen - i changed to Strings.isNullOrEmpty. commit pushed.

**Comment by churrodog** [comment]
thanks for getting this in guys.  I think this is a very useful feature.

**Comment by drcrallen** [comment]
Three days have elapsed. Merging

File: server/src/test/java/org/apache/druid/server/QueryResourceTest.java [hunk 3]
```diff
@@ -401,8 +515,7 @@ public void run()
     startAwaitLatch.await();
 
     Executors.newSingleThreadExecutor().submit(
-        new Runnable()
-        {
+        new Runnable() {
```
Comment (churrodog) [inline server/src/test/java/org/apache/druid/server/QueryResourceTest.java]: formatting, I don't know if we care about internal patches

Comment (jihoonson) [reply server/src/test/java/org/apache/druid/server/QueryResourceTest.java]: Good catch! Yes, it should be fixed.

**Pull request merged**